src/lib.rs

```diff
@@ -178,15 +181,10 @@ fn trie_match_inner(input: ExprMatch) -> Result<TokenStream, Error> {
                 }
                 return #wildcard_idx;
             }
-            let out = *out_checks.get_unchecked(pos) >> 8;
-            if out != 0xffffff {
-                out
-            } else {
-                #wildcard_idx
-            }
+            *out_checks.get_unchecked(pos) >> 8
         })( #expr ) {
             #( #arm, )*
-            _ => unreachable!(),
+            _ => unsafe { std::hint::unreachable_unchecked() },
```

Comment:
```suggestion
            // Safety: A query always matches one of the patterns because
            // all patterns in the input match's AST are expanded. (Even
            // mismatched cases are always captured by wildcard_idx.)
            _ => unsafe { std::hint::unreachable_unchecked() },
```